### ISSUE TITLE
NVDA does not read the accessibility name for a combobox using browse mode

### ISSUE THREAD

**Issue by majornista** [nvaccess/nvda#11188]
### Steps to reproduce:
1. Open [Editable Combobox With List Autocomplete Example from WAI-ARIA Authoring Practices 1.2](https://www.w3.org/TR/wai-aria-practices-1.2/examples/combobox/combobox-autocomplete-list.html)
2. Using browse mode, navigate to State combobox example.

### Actual behavior:
NVDA reads the combobox as, "combo box  collapsed  has auto complete  editable". However, if you were to tab to the combobox so that NVDA switches into forms mode when the combobox receives focus, the combobox will read as, "State  combo box  collapsed  has auto complete  editable  blank".

### Expected behavior:
NVDA should read the accessibility name for the combobox in browse mode as "State  combo box  collapsed  has auto complete  editable".

#### NVDA installed from source:
https://www.nvaccess.org/files/nvda/releases/2020.1/nvda_2020.1.exe

#### NVDA version:
2020.1

#### Windows version:
Windows 10 Enterprise
Version	10.0.18363 Build 18363

#### Name and version of other software in use when reproducing the issue:
Chrome Version 81.0.4044.138 (Official Build) (64-bit),
Firefox 76.0.1 (64-bit)
Edge Chromium Version 81.0.416.77 (Official build) (64-bit)

#### Does the issue still occur after restarting your computer?
Yes.

#### Have you tried any other versions of NVDA? If so, please report their behaviors.
No.

#### If addons are disabled, is your problem still occuring?
Yes.

#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
No.

**Comment by josephsl** [comment]
Hi, how about on Edge 81/83 and Firefox?

**Comment by majornista** [comment]
@josephsl This is a Chrome issue, but it seems to be NVDA's problem. The accessibility name is properly communicated through the accessibility API.

**Comment by josephsl** [comment]
Hi, I see. Is this also the case with other web browsers? If yes, it’s something NVDA people will need to look at, otherwise a collaboration between NV Access/Google/Mozilla/W3C might be helpful. Thanks,

**Comment by jnurthen** [comment]
This reproduces with Firefox too.

**Comment by majornista** [comment]
I can reproduce with Edge Version 81.0.416.77 (Official build) (64-bit), as well.

**Comment by josephsl** [comment]
Hi, how about Firefox 76? I ask for FF results so we can verify that this is truly an NVDA issue across browsers. Thanks.

**Comment by josephsl** [comment]
Hi, I see. I wish I had Marco’s GitHub handle (don’t have it handy at the moment…). CC @JCSTeh

**Comment by majornista** [comment]
Reproduces with Firefox 76.0.1 (64-bit)

**Comment by josephsl** [comment]
Hi, thanks. Do you know of websites other than this one that exhibits the same problem? A few weeks ago either Jamie or Marco noted that web authors are also responsible for something like this. Thanks.

**Comment by jnurthen** [comment]
This will reproduce anywhere the aria 1.2 combobox pattern is used. We (and others) are moving rapidly to this pattern due to its better support than the 1.0 and 1.1 patterns.

**Comment by jcsteh** [comment]
Can't reproduce. The label (State) is on the line above the the combo box. Because it's visible, NVDA doesn't speak it again when you cursor down to the combo box, since that would be duplicate information. This is consistent with how NVDA handles other form fields. However, NVDA does read the label as expected when you press c to quick navigate to the combo box. So, there's no situation in which NVDA doesn't read the label.

**Comment by leonardder** [comment]
I can verify @jcsteh's findings in Firefox 76 and Edge 83, it behaves as it should.

**Comment by Adriani90** [comment]
Yes, pressing c reports the label also in Crhome 83 and Edge Chromium 83. If an user navigates with arrow keys in browse mode, he or she will hear the label above the combo box. This is expected behavior.
Closing as works for me.

**Comment by majornista** [comment]
@Adriani90 and @jcsteh 
I ask that you reconsider closing this bug. 

If you advance beyond the combo box in browse mode, perhaps by using a shortcut to advance to the [Keyboard Support heading]( https://www.w3.org/TR/wai-aria-practices-1.2/examples/combobox/combobox-autocomplete-list.html#kbd_label) just after it, and then `ArrowUp` back to the combo box in browse mode, the Accessibility name is not announced. 

Where this is a problem for us in a real world scenario is in Adobe Experience Manager, where activating a "Search" icon button in the header opens a dialog and focuses a combobox with `aria-label="AEM Search"`. If NVDA fails to shift into forms mode when the Search button is clicked, or when the combobox receives focus, the input is announced as "blank", with no context provided.

**Comment by Adriani90** [comment]
I understand the concern here and I think it is worth to discuss at least following points:
1. NVDA should announce the lable when the focus is redirectioned to the lement after pressing enter anwhere else and
2. NVDA could announce the label of combo boxes and edit fields in line with the element and not render the label on the line above in browse mode. The down side of this would be that the user cannot navigate the label by character or word, but this could be acomplished i.e. by pressing nvda.l or nvda+shift+o twice to read the current line / current object by character.

I am reopening.

**Comment by jcsteh** [comment]
> If you advance beyond the combo box in browse mode, perhaps by using a shortcut to advance to the [Keyboard Support heading](https://www.w3.org/TR/wai-aria-practices-1.2/examples/combobox/combobox-autocomplete-list.html#kbd_label) just after it, and then `ArrowUp` back to the combo box in browse mode, the Accessibility name is not announced.

Sure. Nor is a heading announced when you move back up through a paragraph.

> Where this is a problem for us in a real world scenario is in Adobe Experience Manager, where activating a "Search" icon button in the header opens a dialog and focuses a combobox with `aria-label="AEM Search"`. If NVDA fails to shift into forms mode when the Search button is clicked, or when the combobox receives focus, the input is announced as "blank", with no context provided.

IMO, that's the real bug here; i.e. the failure to switch to focus mode, not where the label gets rendered. However, that's very different to what was originally specified.

>     2. NVDA could announce the label of combo boxes and edit fields always in line with the element in browse mode and not render the label on the line above. The down side of this would be that the user cannot navigate the label by character or word, but this could be acomplished i.e. by pressing nvda.l or nvda+shift+o twice to read the current line / current object by character.

There are other downsides to that, such as the loss of formatting and semantics (links, images, etc.) within the label. I covered this in a [blog post](https://www.jantrid.net/2015/12/woe-aria-surprisingly-but-ridiculously.html); see the section "Ignore the label element, obviously. Duh.". This is precisely why we didn't do this in the first place.

**Comment by majornista** [comment]
@jcsteh I agree that the failure to switch to focus mode may be the real problem here.

**Comment by Adriani90** [comment]
@majornista could you please edit the issue title and issue description including your real problem? If possible, Ideally we need to have a minimal test case where NVDA shows the bug in practice. That means i.e. pressing on the button, NVDA's virtual cursor in browse mode jumps to a combo box but NVDA does not switch to focus mode and so the label is not announced.

**Comment by majornista** [comment]
@Adriani90 and @jcsteh I'm having a difficult time reproducing an example where NVDA fails to shift into focus mode when the combobox receives focus using NVDA Version 2023.2 (2023.2.0.29051). Also, in the few examples of search dialogs within Adobe Experience Manager I tested, while NVDA take a few seconds to shift into focus mode when the combobox search field within the dialog receives focus, when NVDA did shift into focus mode, it announced the accessibility name with the input.

**Comment by Adriani90** [comment]
@majornista thanks very much for this update. NVDA improved gradually during last years so it is certainly possible that this issue does not exist anymore. I am closing it for now until we get another test case to reproduce reliably.
